pkg/skaffold/docker/auth.go

```diff
@@ -61,13 +62,22 @@ type AuthConfigHelper interface {
 
 type credsHelper struct{}
 
-func (credsHelper) GetAuthConfig(registry string) (types.AuthConfig, error) {
+func loadDockerConfig() (*configfile.ConfigFile, error) {
 	cf, err := config.Load(configDir)
 	if err != nil {
-		return types.AuthConfig{}, errors.Wrap(err, "docker config")
+		return nil, errors.Wrap(err, "docker config")
 	}
 
-	gcp.AutoConfigureGCRCredentialHelper(cf, registry)
+	gcp.AutoConfigureGCRCredentialHelper(cf)
+
+	return cf, nil
+}
+
+func (credsHelper) GetAuthConfig(registry string) (types.AuthConfig, error) {
+	cf, err := loadDockerConfig()
+	if err != nil {
+		return types.AuthConfig{}, err
```

Comment:
This means a broken config.json will cause skaffold to abort.  Maybe we should log an error, but continue on?

Reply:
1/ docker-credential-gcloud should be favored. See https://github.com/GoogleCloudPlatform/docker-credential-gcr "For normal development setups, users are encouraged to use gcloud auth configure-docker, instead"
2/ In the docker build code, we ignore the error, like the Docker CLI does it.
3/ We can't just configure it for `gcr.io` when `GetAllAuthConfigs()` is used. When `GetAuthConfig` is used, configuring the helper for *.gcr.io has the exact same effect